### ISSUE TITLE
feature: ability to measure throughput

### ISSUE THREAD

**Issue by steven-varga** [p-ranav/criterion#5]
Hi, great work, and crisp coding style! I have a question: I am trying to add throughput measurement functionality -- so I could use your work in [H5CPP](https://github.com/steven-varga/h5cpp)  and other HDF5 performance measurement code; and wondering what is the best way to tackle this problem?

Would it be possible to return a `long double` from the `criterion::benchmark_config::FN = std::function<...>`? Or perhaps some other suggestion? 
By no means am I  here distract you from your work -- but I am already a consumer to you `argparse` and `criterion` is very similar how I would organise code base of this type; and most importantly `criterion` can provide almost what IO benchmarking would need. 

I understand if the answer is no, but I though it is better to ask. 
best wishes: steven

**Comment by p-ranav** [comment]
Hello! Thanks for your comments and request. 

I'm all for supporting additional measurement functionality, e.g., throughput. 

What would the `long double` return value represent? I suppose you would use such a return value [here](https://github.com/p-ranav/criterion/blob/master/include/criterion/details/benchmark.hpp#L167)? Feel free to create a PR if you already have some changes you'd like to go over.

**Comment by steven-varga** [comment]
Thank you for getting back to me! The `long double` return value would represent the amount of data transferred in bytes. This way the throughput can be calculated as the execution time is already available.  This is a common pattern in performance measurement -- and I think it is a useful addition to your excellent library.

I tried to modify the suggested line, unfortunately it didn't quite work out. So I ended up writing a yet-another-benchmark library, with slightly different syntax and no macros. The rational behind this move was we want the result in HDF5 format -- this is easy to add to your library --; but also want it to be MPI compatible: this requires more work.  

sorry about the noise:
steve
